client/system/detect_cloud/gce.go

```diff
@@ -0,0 +1,24 @@
+package detect_cloud
+
+import (
+	"net/http"
+)
+
+func detectGCE() string {
+	r, err := http.NewRequest("GET", "http://metadata.google.internal", nil)
+	if err != nil {
+		return ""
+	}
+	r.Header.Add("Metadata-Flavor", "Google")
+
+	resp, err := hc.Do(r)
+	if err != nil {
+		return ""
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusOK {
+		return "Google Compute Engine"
```

Comment:
```suggestion
		return "Google Cloud Platform"
```

is this generic GCP better here? Since we also use `aws` and not `ec2`. The whole filename then, really.